packages/typespec-python/test/azure/generated/azure-mgmt-spheredpg/azure/mgmt/spheredpg/models/_models.py

```diff
@@ -142,8 +144,28 @@ def __init__(self, mapping: Mapping[str, Any]):
         """
 
     def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useless-super-delegation
+        flatten_map = {k: kwargs.pop(k, None) for k in self._flatten_items}
         super().__init__(*args, **kwargs)
-
+        for k, v in flatten_map.items():
+            if v is not None:
+                setattr(self, k, v)
+
+    def __getattr__(self, name: str) -> Any:
+        if name in self._flatten_items:
+            if self.properties:
+                return getattr(self.properties, name)
+            return None
+        raise AttributeError(f"'{self.__class__.__name__}' object has no attribute '{name}'")
+
+
+    def __setattr__(self, key: str, value: Any) -> None:
+        if key in self._flatten_items:
+            if self.properties is None:
+                class_type = self._attr_to_rest_field["properties"]._class_type
+                self.properties = class_type()
```

Comment:
super small nit: I feel like it would look better if you just did `self.properties = self._attr_to_rest_field["properties"]._class_type`